de.tu_bs.cs.isf.e4cf.compare.data_structures/src/de/tu_bs/cs/isf/e4cf/compare/data_structures/io/writter/JavaWriterUtil.java

```diff
@@ -0,0 +1,648 @@
+
+package de.tu_bs.cs.isf.e4cf.compare.data_structures.io.writter;
+
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import de.tu_bs.cs.isf.e4cf.compare.data_structures.interfaces.Node;
+import de.tu_bs.cs.isf.e4cf.compare.data_structures.io.reader.JavaNodeTypes;
+
+import com.github.javaparser.ast.body.*;
+import com.github.javaparser.ast.expr.*;
+import com.github.javaparser.ast.nodeTypes.*;
+import com.github.javaparser.ast.stmt.*;
+import com.github.javaparser.ast.type.*;
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.*;
+
+/**
+ * 
+ * @author Serkan Acar
+ * @author Pascal Blum
+ * @author Paulo Haas
+ * @author Hassan Smaoui
+ *
+ */
+public class JavaWriterUtil {
+	/**
+	 * Converts a framework node and it's children to a JavaParser node
+	 * 
+	 * @param n Framework node
+	 * @param p is the parent node of n's equivalent JavaParserNode. This can be
+	 *          null.
+	 * @return n as JavaParser node
+	 * @throws UnsupportedOperationException if an attribute, node type etc. is not
+	 *                                       supported.
+	 */
```

Comment:
Methods in this class need more inline comments. It is hard to understand what all these cases are.